File: .NET Core Web API including SOLID and Design Patterns/SurveyAPI/Models/ControllerModels/CreateSurvey.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.Linq;
using System.Threading.Tasks;

namespace SurveyAPI.Models
{
    public class CreateSurvey
    {
        [Required]
        public string Name { get; set; }
        [Required]
        public DateTime StartDate { get; set; }
        [Required]
        public DateTime EndDate { get; set; }
        [Required]
        public List<CreateQuestions> Questions { get; set; }
    }
}

File: Angular Web APP/survey-webapp/src/app/models/create-survey.ts
export class CreateSurvey{
    public name:string;
    public startDate:Date;
    public endDate:Date;
    public questions:CreateQuestion[];
}
export class CreateQuestion {
    public questionText: string;
    public offeredAnswers : CreateOfferedAnswer[]
}
export class CreateOfferedAnswer{
    public text:string;
}
